arelle/CntlrWinMain.py

```diff
@@ -816,15 +819,16 @@ def fileOpenFile(self, filename, importToDTS=False, selectTopView=False):
         if filename:
             for xbrlLoadedMethod in pluginClassMethods("CntlrWinMain.Xbrl.Open"):
                 filename = xbrlLoadedMethod(self, filename) # runs in GUI thread, allows mapping filename, mult return filename
-            filesource = None
+            filesource, entrypointFiles, success = parseEntrypointFileInput(self, filename, autoSelect=False)
+            if not success:
+                return
             # check for archive files
-            filesource = openFileSource(filename, self,
-                                        checkIfXmlIsEis=self.modelManager.disclosureSystem and
-                                        self.modelManager.disclosureSystem.validationType == "EFM")
             if filesource.isArchive:
                 if not filesource.selection and not filesource.isReportPackage: # or filesource.isRss:
                     from arelle import DialogOpenArchive
                     filename = DialogOpenArchive.askArchiveFile(self, filesource)
+                    if filename is not None:
+                        entrypointFiles.append({"file": filename})
```

Comment:
Should the user be able to select additional files if parseEntrypointFileInput populated entrypointFiles? If yes, do we need to make sure they haven't selected duplicates from parseEntrypointFileInput?